### ISSUE TITLE
invalidateSession "broken" in latest release version of ember.

### ISSUE THREAD

**Issue by cibernox** [simplabs/ember-simple-auth#593]
Coming from here: https://github.com/emberjs/ember.js/pull/11864#issuecomment-125213627

**tl; dr;**

The latest version of ember stable (not yet tagged but already available using `components/ember#release` and will be released in 1.13.6) includes a bugfix in some memory leaks. 

One of the fixes consists in unregister test helpers when the app is destroyed. However this causes an error in `invalidateSession` when you have an `afterEach` hook like this:

```js
  afterEach: function() {
    invalidateSession();
    Ember.run(application, 'destroy');
  }
```

It turns out that `invalidateSession` is asynchronous (BTW, why?) the destruction of the app is synchronous because happens inside an `Ember.run`, so by the time the session is invalidated the `wait()` helper is not available anymore.

Doing `Ember.run(invalidateSession);` worked for me, although @stefanpenner suggested to chain the invalidation of the session like this:

`invalidateSession().then(() => Ember.run(application, 'destroy'))`

It's a bit verbose but I'm ok with it if it has to be done that way, as long as it's explained in the documentation.

I'd also like to know why the invalidation of the session is async, because perhaps the helper can be run-loop aware and allow to keep being used without chaining.

**Comment by marcoow** [comment]
`invalidateSession` asynchronous because the underlying authenticator implementation might be asynchronous. When you're using the ephemeral session store you shouldn't need the `invalidateSession` call at all though as every time the application restarts you'll have a new session with a clean store.

**Comment by cibernox** [comment]
I don't remember why I added this, but I remember some tests failing without it. I'll investigate it.

**Comment by marcoow** [comment]
closing - please reopen if the problem persists
